src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -145,11 +145,13 @@ private unsafe void RemoveNodeFromDatabase(TransactionOperationContext context,
                     items.DeleteByKey(loweredKey);
                     return;
                 }
-
+                //TODO: Remove those 3 lines? i think it is already removed at this point
```

Comment:
No, this is handled when you are deletion the database entirely, but not when you are removing the node.